### DISCUSSION TITLE
Options - Toolbar - Button to open Lua and Bash scripts

### DISCUSSION THREAD

**Discussion by advice2020** [doublecmd/doublecmd#1226] [Suggestions]
Currently if you want to modify a Lua or Bash script that is linked to an existing `cm_ExecuteScript` / External Command, you have to manually navigate to your scripts folder and find that specific script file linked to the tool button.

Suggestng to add a new button that if a path is detected in the first line in parameter typing field pane, for example....
`lua-script-path/lua-script-name.lua`
or
`bash-script-path/bash-script-name.sh`
then this button would open up that script file in associated text editor.

Thank You

**Comment by Skif-off** [comment]
2.) Sometimes I use this button
```
internal command: cm_ExecuteScript
params: %"0%p0
```
also it can be
```
internal command: cm_ExecuteScript
params:
%"0%p0
%"0%D
```
(i.e. for the script under the cursor): what should the program do in this case?

I think most users use one or two (automation + WDX; I know two exceptions ([marker](https://github.com/j2969719/doublecmd-plugins/tree/master/utils/marker), [tags](https://github.com/j2969719/doublecmd-plugins/tree/master/utils/tags)), but I don't know if anyone is using these) directories for scripts and opening this folder in the file panel shouldn't be much of a problem :)

3.) To the top? Any script can also be an additional parameter.

4.) It can be script like Bash, Python, and so on (they can be used even without an extension): make them executable and add a [shebang](https://en.wikipedia.org/wiki/Shebang_(Unix)).
Of course, such scripts must be used as external commands.

P.S. Many internal commands accept parameters as "parameter=value" instead of just "value", drag-and-drop does not combine well with this.

**Comment by advice2020** [comment]
@Skif-off
Thank you for your response, I don't get a ton of responses on this site so I didn't even realize you had responded.
Just letting you know that I updated this post so your response will no longer apply.
Up to you if want to delete or update your post accordingly.  (I will delete this/my reply after you do)
Sorry for the confusion just wanted to simplify this request as much as I can.
Thank you again for responding

**Comment by advice2020** [comment]
Ahh, still surprised no users would have found this one useful.
